src/nativegasestimator.rs

```diff
@@ -283,19 +290,16 @@ async fn collect_rewards<T: Transport + Send + Sync>(
                 )
                 .await?;
 
-            if fee_history.reward.is_none() {
-                break;
-            }
-
-            let fee_history_reward = fee_history.reward.unwrap();
-            for reward in &fee_history_reward {
-                for i in 0..=params.max_reward_percentile {
-                    let reward = reward[i].low_u64();
-                    if reward > 0 {
-                        rewards.push(reward);
-                    }
-                }
-            }
+            let fee_history_reward = fee_history
+                .reward
+                .ok_or_else(|| anyhow!("missing reward in response"))?;
```

Comment:
I think this would be equivalent (but it would also require `anyhow::Context` to be in scope):
```suggestion
            let fee_history_reward = fee_history
                .reward
                .with_context(|| "missing reward in response"))?;
```

Reply:
Even better:
```suggestion
            let fee_history_reward = fee_history
                .reward
                .context("missing reward in response")?;
```

Reply:
AFAIU `.context(&str)` doesn't need to allocate, since internally it only adds to the error chain if there is an error.

This is different if you need formatting though:
```rust
// prefer this:
some_result().with_context(|| format!("some {details}"))?;

// over this:
some_result().context(format!("some {details}"))?;
```

Reply:
Ahh, I see. Makes sense. 👍

Reply:
Neat. I didn't know this was also implemented for Option.